File: backend/src/main/java/com/tracepilot/api/DTO/Request/AuditRequest.java
package com.tracepilot.api.DTO.Request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import com.tracepilot.api.Enums.AuditInputSource;

public record AuditRequest(
        @NotBlank(message = "Raw trace cannot be empty")
        @Size(max = 80000, message = "Trace exceeds 80,000 characters")
        String rawTrace,
        
        @Size(max = 150)
        String title,
        
        @Size(max = 150)
        String repoName,
        
        @Size(max = 30)
        String agentTool,
        
        AuditInputSource inputSource
) {}

File: frontend/src/schemas/audit-request.ts
import { z } from "zod";
import { AuditInputSourceSchema, AgentToolRequestSchema } from "./audit";

export const AuditRequestSchema = z.object({
  rawTrace: z.string().min(1).max(80000),
  title: z.string().max(150).optional(),
  repoName: z.string().max(150).optional(),
  agentTool: AgentToolRequestSchema,
  inputSource: AuditInputSourceSchema.optional(),
});
export type AuditRequest = z.infer<typeof AuditRequestSchema>;